specs/lib/gltfPipelineSpec.js

```diff
@@ -114,13 +114,47 @@ describe('gltfPipeline', function() {
         });
     });
 
+    it('will write sources from JSON', function(done) {
+        var spy = spyOn(fs, 'writeFile').and.callFake(function(file, data, callback) {
+            callback();
+        });
+        var options = {
+            createDirectory : false,
+            embed : false
+        };
+        readGltf(gltfPath, options, function(gltf) {
+            processJSON(gltf, outputPath, options, function() {
+                process.nextTick(function() {
+                    expect(spy.calls.count()).toEqual(4);
```

Comment:
@JoshuaStorm Can you add a check in these tests making sure that source actually does get updated? Maybe make a change to it while it is in pipeline extras then decode the uri after and compare.